### PR TITLE
BROKER: Select Student By Id

### LINKED ISSUE #96: BROKER: Select Student By Id

### PR DESCRIPTION
**Pull request by muhammadmnorouzi**
closes #96

### CODE REVIEW

File: SCMS.Services.Api/Brokers/Storages/IStorageBroker.Student.cs [hunk 1]
```diff
@@ -10,5 +11,7 @@ namespace SCMS.Services.Api.Brokers.Storages
     public partial interface IStorageBroker
     {
         public ValueTask<Student> InsertStudentAsync(Student student);
+
```
Comment (ShriHumrudha) [inline SCMS.Services.Api/Brokers/Storages/IStorageBroker.Student.cs:14]: Remove the extra line, these can be together as it is one line code.

**Review by ShriHumrudha — CHANGES_REQUESTED** [review]

File: SCMS.Services.Api/Brokers/Storages/StorageBroker.Student.cs [hunk 2]
```diff
@@ -2,6 +2,8 @@
 // Copyright (c) Signature Chess Club & MumsWhoCode. All rights reserved.
 // -----------------------------------------------------------------------
 
+
```
Comment (ShriHumrudha) [inline SCMS.Services.Api/Brokers/Storages/StorageBroker.Student.cs:5]: Remove this extra line

File: SCMS.Services.Api/Brokers/Storages/StorageBroker.Student.cs [hunk 3]
```diff
@@ -25,7 +27,15 @@ public async ValueTask<Student> InsertStudentAsync(Student student)
...
+      
+        public async ValueTask<Student> SelectStudentByIdAsync(Guid studentId)
+        {
+            using var broker = new StorageBroker(this.configuration);
+            broker.ChangeTracker.QueryTrackingBehavior = QueryTrackingBehavior.NoTracking;
 
+            return await broker.Students.FindAsync(studentId);
+        }
+      
         public IQueryable<Student> SelectAllStudents() => this.Students.AsQueryable();
```
Comment (ShriHumrudha) [inline SCMS.Services.Api/Brokers/Storages/StorageBroker.Student.cs:39]: Select All should be implemented first then followed by SelectById

**Review by ShriHumrudha — CHANGES_REQUESTED** [review]

File: SCMS.Services.Api/Brokers/Storages/StorageBroker.Student.cs (hunk 3 above)
Comment (muhammadmnorouzi) [reply SCMS.Services.Api/Brokers/Storages/StorageBroker.Student.cs]: Thank you for you reviews.

**Review by ShriHumrudha — APPROVED** [review]

**Pull request merged**